File: src/main/java/com/lczq/dbquery/service/ConnectionDB.java
package com.lczq.dbquery.service;

import com.lczq.dbquery.entities.DataSource;
import com.lczq.dbquery.entities.QueryConfig;
import com.lczq.dbquery.entities.QueryResult;
import org.springframework.stereotype.Service;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.ResultSetMetaData;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.HashMap;
import java.util.Map;

@Service
public class ConnectionDB
{

    public QueryResult executeSQL(QueryConfig queryConfig, DataSource dataSource)
            throws ClassNotFoundException, SQLException
    {
        String dbUrl = dataSource.getUrl();
        String user = dataSource.getUsername();
        String pass = dataSource.getPassword();
        String jdbcDriver = dataSource.getDriver();
        String sql = queryConfig.getQuerySql();

        QueryResult rsList = new QueryResult();
        Map<String, Object> rsMap;
        Connection conn;
        Statement stmt;

        Class.forName(jdbcDriver);
        conn = DriverManager.getConnection(dbUrl, user, pass);

        stmt = conn.createStatement();
        ResultSet rs = stmt.executeQuery(sql);
        ResultSetMetaData rsmd = rs.getMetaData();
        int numberOfColumns = rsmd.getColumnCount();
        while (rs.next()) {
            rsMap = new HashMap<>(numberOfColumns);
            for (int i = 1; i < numberOfColumns + 1; i++) {
                rsMap.put(rsmd.getColumnName(i), rs.getObject(i));
            }
            rsList.setOneMap(rsMap);
        }
        rs.close();
        stmt.close();
        conn.close();

        return rsList;
    }
}

File: backend/src/main/kotlin/com/github/wgzhao/dbquery/service/ConnectionDB.kt
package com.github.wgzhao.dbquery.service

import com.github.wgzhao.dbquery.entities.DataSources
import com.github.wgzhao.dbquery.entities.QueryConfig
import org.springframework.stereotype.Service
import java.sql.Connection
import java.sql.DriverManager
import java.sql.SQLException
import java.sql.Statement

@Service
class ConnectionDB {
    @Throws(ClassNotFoundException::class, SQLException::class)
    fun executeSQL(queryConfig: QueryConfig, dataSource: DataSources): MutableList<MutableMap<String?, Any?>?> {
        return executeSQL(queryConfig.querySql, dataSource)
    }

    @Throws(ClassNotFoundException::class, SQLException::class)
    fun executeSQL(querySql: String?, dataSource: DataSources): MutableList<MutableMap<String?, Any?>?> {
        val dbUrl = dataSource.url
        val user = dataSource.username
        val pass = dataSource.password
        val jdbcDriver = dataSource.driver

        val result: MutableList<MutableMap<String?, Any?>?> = ArrayList()

        Class.forName(jdbcDriver)
        DriverManager.getConnection(dbUrl, user, pass).use { conn ->
            conn.createStatement().use { stmt ->
                val rs = stmt.executeQuery(querySql)
                val rsmd = rs.metaData
                val numberOfColumns = rsmd.columnCount
                while (rs.next()) {
                    val rsMap: MutableMap<String?, Any?> = HashMap(numberOfColumns)
                    for (i in 1..numberOfColumns) {
                        rsMap[rsmd.getColumnLabel(i)] = rs.getObject(i)
                    }
                    result.add(rsMap)
                }
                rs.close()
            }
            conn.close()
        }

        return result
    }
}
